examples/all-clusters-app/all-clusters-common/include/oven-operational-state-delegate.h

```diff
@@ -0,0 +1,118 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace OvenCavityOperationalState {
+
+// This is an application level delegate to handle operational state commands according to the specific business logic.
+class OvenCavityOperationalStateDelegate : public OperationalState::Delegate
+{
+private:
+    const Clusters::OperationalState::GenericOperationalState mOperationalStateList[8] = {
```

Comment:
Why is this of size 8, but there are only 4 elements

Reply:
I agree. The getOperationalStateAtIndex function isn't going to work properly with this partially filled array. Code is updated with this change.